tests/xfa/Makefile

```
@@ -2,8 +2,9 @@ include ../Makefile.tests_common
 
 include $(RIOTBASE)/Makefile.include
 
-all: static-test
-static-test: $(ELFFILE)
+BUILDDEPS += static-test
+
+static-test:
```

I tested with the following just to make sure it was as expected

```suggestion
ifeq ($(BUILD_IN_DOCKER),1)
all: ..in-docker-container
else
ifeq ($(INSIDE_DOCKER),1)
  $(info #### indocker)
  all: static-test
else
  $(info #### no docker)
  all: static-test
endif
endif
```

```Launching build container using image "riot/riotbuild:latest".
docker run --rm --tty --user $(id -u) -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/home/francisco/workspace/RIOT:/data/riotbuild/riotbase:delegated' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'        -w '/data/riotbuild/riotbase/tests/xfa/' 'riot/riotbuild:latest' make     -j2 all
#### indocker
Building application "tests_xfa" for "native" with MCU "native".

"make" -C /data/riotbuild/riotbase/boards/native
"make" -C /data/riotbuild/riotbase/core
"make" -C /data/riotbuild/riotbase/boards/native/drivers
"make" -C /data/riotbuild/riotbase/cpu/native
"make" -C /data/riotbuild/riotbase/cpu/native/periph
"make" -C /data/riotbuild/riotbase/cpu/native/stdio_native
"make" -C /data/riotbuild/riotbase/drivers
"make" -C /data/riotbuild/riotbase/drivers/periph_common
"make" -C /data/riotbuild/riotbase/sys
"make" -C /data/riotbuild/riotbase/sys/auto_init
"make" -C /data/riotbuild/riotbase/sys/test_utils/interactive_sync
   text    data     bss     dec     hex filename
  23604    4714   47748   76066   12922 /data/riotbuild/riotbase/tests/xfa/bin/native/tests_xfa.elf

```

---

```suggestion
  ifeq ($(BUILD_IN_DOCKER),1)
all: ..in-docker-container
else
all: static-test
endif
```

But this should work the same as above.